WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.h

```diff
@@ -154,6 +158,36 @@ typedef NS_ENUM(NSUInteger, FBUIInterfaceAppearance) {
  */
 - (nullable NSNumber *)fb_getAppearance;
 
+#if !TARGET_OS_TV
+/**
+ Allows to set a simulated geolocation coordinates.
+ Only works since Xcode 14.3/iOS 16.4
```

Comment:
Hm.... this set command kept the location while the WDA process stopped. After re-start the WDA process, the device responded did not have the simulated location (got an error against get simulated location method).
The DELETE simulated endpoint did not help. Tuen the location service off, and re-on also did not help. In my real device testing, only restarting the device brought the location back to normal.

Can you address something like: "if the WDA (XCTest) process ended without deleting the simulated location endpoint call, the location could remain until restarting the device"?

Reply:
Did a few more testing in a real device.

- setSimulatedLocation applied the location immediately rather than the location service via appium-ios-device.
- clearSimulatedLocationWithReply is more reliable to delete the simulated location than reset it via appium-ios-device
- getSimulatedLocationWithReply returns proper location by the setSimulatedLocation rather than current location info in the deviceInfo endpoint
- Resetting the device location without calling clearSimulatedLocationWithReply, and reset it via the location service, both needed to reboot the device to reset the location perfectly

So, for via the xcuitest driver for real devices, basically using this PR endpoint is better, but probably can add a capability to call "clearSimulatedLocationWithReply" in every deleting a session, or we call it by default.

Reply:
I just wanted to add these as extensions anyway to not break the backward compatibility
Eventually /location endpoints are not part of W3C standard